### DISCUSSION TITLE
Enhancements for the Range support

### DISCUSSION THREAD

**Discussion by WeihanLi** [dotnet/csharplang#6132] [Q&A] [answered]
Currently, when we use Range with an expression, we would get an error likes `Operator '+' cannot be applied to operands of type 'int' and 'Range'`, `CS0019`

![image](https://user-images.githubusercontent.com/7604648/168463321-81931619-9639-4ea2-b934-7f413ed65312.png)

Could we make it more simple to use, wondering if it's possible to let the range operator `..` priority lower than other operations so that we could use it without a brace

**Comment by kyoyama-kazusa** [comment, marked as answer]
Just add a brace: `(i + 1)..`

See docs:

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-precedence

Picture:

![image](https://user-images.githubusercontent.com/23616315/168463688-6df143e3-344b-4f41-b233-be7c5e03427e.png)

**Reply by WeihanLi** [reply]
Thanks, I know this, I'm wondering if it's possible to let it work without the brace

**Reply by kyoyama-kazusa** [reply]
@WeihanLi Hello.

`(i + 1)..` and `i + 1..` are two different expressions. For the reference of the operator precedence, we know that `i + 1..` means a variable `i` adds a range expression `1..`, which is incorrect. We know that an integer cannot add to a range expression because of not being defined.

Actually, I prefer the brace-styled syntax rather than `i + 1..` because `i + 1..` is not clear enough. Generally, a range expression can omit start index or end index, or both, which means the expression will be like:

* `1..`
* `..^2`
* `..`

Which makes the range syntax be more like a unary expression. A unary expression always has a higher priority than the binary one, for example, `++x` has a higher priority than `x + 1`. If we assign a range expression with a lower precedence, the rule will be broken.

For me, if without a brace, the readability will be slow down because I may often misread the expression, to get an incorrect expression `i + (1..)` instead of `(i + 1)..` because IDE always inserts a whitespace into the place between plus sign `+` and literal one `1`.

**Comment by svick** [comment]
> Could we make it more simple to use

What is the suggested solution? Since changing the precedence would be a breaking change, I don't think it's likely that code like `p[i+1..]` will ever compile. Would a quick fix that adds the parentheses for you be enough?

**Comment by 333fred** [comment]
The precedence of `..` was very intentionally chosen to bind tighter than `+`. `..` is usually written without spaces, and `+` is usually written with spaces. We feel that it would be confusing to the reader if the thing with spaces binds tighter than the thing without them, as the rest of C# doesn't work this way.
